Arm: Add soc support for rda5981a

liliu-l opened the pull request.

liliu-l committed:
arm: Add support for rda5981a

RDA5981A: ARM Cortex-M4 Wireless MCU (64kB ROM, 32kB cache)

Signed-off-by: Li Liu <li.liu@linaro.org>

nashif commented:
Can you add some more details in the commit message what rda5991h is exactly, a link maybe, which vendor is this, etc... Do not expect everyone to know what this board is.

liliu-l commented:
commit message updated.

liliu-l committed:
boards: Add board definition for rda5981a

RDA5981A: ARM Cortex-M4 Wireless MCU (64kB ROM, 32kB cache)

Signed-off-by: Li Liu <li.liu@linaro.org>

JammyZhou reviewed (commented):
How about change the folder name "96b_rda" to "96b_bluesky" since the board is announced as BlueSky at the Linaro BUD17 connect? Accordingly, rename the '96b_rda_defconfig' file to '96b_bluesky_defconfig' as well as other macro names, etc

JammyZhou commented on boards/arm/96b_rda/96b_rda_defconfig:
```diff
@@ -0,0 +1,27 @@
+CONFIG_ARM=y
+CONFIG_BOARD_96B_RDA=y
+CONFIG_SOC_SERIES_RDA5991=y
```
Should it be changed to CONFIG_SOC_SERIES_RDA5981?

liliu-l commented:
changed 96b_rda to 96b_bluesky

liliu-l committed:
gpio: Add GPIO driver for rda5981a (ARM Cortex-M4 Wireless MCU)

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l commented:
resolved conflicts

MaureenHelm commented on arch/arm/soc/rda_5981/Kconfig:
```diff
@@ -0,0 +1,23 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config SOC_FAMILY_RDA5981
```
This is a really specific name for an SoC family. Is there a more general name you could use that could encompass additional SoCs in the family?

liliu-l replied:
There is not yet a more generic name. 5981 is the SoC  famliy name, which includes 5981a now.

MaureenHelm commented on arch/arm/soc/rda_5981/Kconfig:
```diff
@@ -0,0 +1,23 @@
+config SOC_RDA_DEBUG
+	bool
+	prompt "Use for RDA SOC DBG"
+	default n
```
Can you please explain this more? Is this hardware debug?

liliu-l replied:
It is not hardware debug.If select Y here, you can  print more debugging messages.

MaureenHelm commented on arch/arm/soc/rda_5981/Kconfig.defconfig:
```diff
@@ -0,0 +1,17 @@
+	default 0x100080
+
+config FLASH_BASE_ADDRESS
+	default 0x018001000
```
sram and flash base addresses should come from DTS

liliu-l replied:
DTS is not  defined currently, will add DTS later.

MaureenHelm commented on arch/arm/soc/rda_5981/5981a/soc_registers.h:
```diff
@@ -0,0 +1,109 @@
+};
+
+/* include register mapping headers */
+struct uart_rda5981a {
```
Does this SoC have CMSIS headers that we could use instead?

liliu-l replied:
Yes, rda5981 has CMSIS headers, will add it later.

MaureenHelm commented on arch/arm/soc/rda_5981/5981a/soc_cfg.c:
```diff
@@ -0,0 +1,250 @@
+}
+
+/* Read ADC value */
+unsigned short rda_ccfg_adc_read(unsigned char ch)
```
This should go in a separate ADC driver

liliu-l replied:
will move this to ADC driver.

MaureenHelm commented on arch/arm/soc/rda_5981/5981a/soc_cfg.c:
```diff
@@ -0,0 +1,250 @@
+}
+
+/* Config UART0 RX pin */
+static inline int rda_ccfg_gp26(void)
```
Do you have a UART driver?

liliu-l replied:
yes, we have UART driver, will push UART driver later.

MaureenHelm commented on drivers/gpio/gpio_rda5981a.c:
```diff
@@ -0,0 +1,230 @@
+		gpio_rda5981a_write(dev, GPIO_ACCESS_BY_PIN,
+					GPIO_PIN3, !toggle);
+		k_sleep(1000);
+		toggle = toggle ? 0 : 1;
```
Why is this splash thread in the gpio driver? It looks more like a sample or test application

liliu-l replied:
It is s test case.  Have removed it.

liliu-l replied:
It is s test case, have removed it.

MaureenHelm reviewed (changes requested):
Need to add board docs and DTS.

It also looks like you need to rebase to the latest arm branch.

liliu-l committed:
boards: Add more info for SOC_RDA_DEBUG

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
serial: added UART driver for rda5981a (ARM Cortex-M4 Wireless MCU)

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
gpio: Remove gpio test case from gpio_rda5981a.c

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
pinmux: add pinmux driver for rda5981a (ARM Cortex-M4 Wireless MCU)

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
pinmux: add pinmux driver for rda5981a (ARM Cortex-M4 Wireless MCU)

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
bluesky: Revert "Remove gpio test case from gpio_rda5981a.c"

This reverts commit c3b7fabd741b9b219d3d55aabe0c40e5b669fde8.

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
bluesky: Revert "Remove gpio test case from gpio_rda5981a.c"

This reverts commit c3b7fabd741b9b219d3d55aabe0c40e5b669fde8.

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
bluesky: Revert "Add more info for SOC_RDA_DEBUG"

This reverts commit d7ca792abb92d0383a7198d46800a7f170f3cebe.

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
bluesky: Revert "Add more info for SOC_RDA_DEBUG"

This reverts commit d7ca792abb92d0383a7198d46800a7f170f3cebe.

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
bluesky : fix compile errors.

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
bluesky : fix compile errors.

Signed-off-by: Li Liu <li.liu@linaro.org>

galak commented:
Can you rebase this against recent arm branch, and address the issues that shippable reported

liliu-l committed:
watchdog: Add WDT driver for rda5981a (ARM Cortex-M4 Wireless MCU)

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
driver: add sdmmc driver for rda5981a (ARM Cortex-M4 Wireless MCU)

Signed-off-by: Li Liu <li.liu@linaro.org>

liliu-l committed:
dts: 96b_bluesky: Add dts support for rda5981a

RDA5981A: ARM Cortex-M4 Wireless MCU (64kB ROM, 32kB cache)

Signed-off-by: Li Liu <li.liu@linaro.org>

fabled committed:
arm: corex_m: add byte/half-word sized memory accessors

The byte ones are required for ns16550 uart driver which is
present on some arm socs. Add half-word ones for completeness.

Signed-off-by: Timo Teräs <timo.teras@iki.fi>
Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

fabled committed:
serial/uart_ns16550: keep LCR settings when programming DLAB

The LCR register specifies the format of the asynchronous data
communication exchange. Keep the data/stop bit and parity settings
intact during baud rate change.

Signed-off-by: Timo Teräs <timo.teras@iki.fi>

fabled committed:
serial/uart_ns16500: poll THRE to determine when data can be written

TEMT is Transmitter Empty bit which is set only when the full FIFO
is empty. It makes sense to poll for THRE (Transmitter Holding
Register Empty) which is set when UART can buffer new character
for transmission (there is room in FIFO). This allows using the
FIFO in full.

Signed-off-by: Timo Teräs <timo.teras@iki.fi>

d3zd3z committed:
arm: Add partition table to hexiwear

Use the same partition table as is used on the frdm_k64 board.

Signed-off-by: David Brown <david.brown@linaro.org>

ydamigos committed:
arch: soc: st_stm32: Remove unused enumeration from F4, F3 SoCs

The enumerations stm32{f3,f4}x_pin_config_mode aren't used any
more. This patch removes them.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

Mani-Sadhasivam committed:
dts: arm: 96b_carbon: add I2C_2 interface

This patch enables I2C_2 support for 96b_carbon. Without
this patch, trying to build I2C_2 will result in build
error.

Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>

MaureenHelm committed:
dts: Fix uart irq info for frdm_k64f and hexiwear_k64

Fixes uart irq info in the device tree fixup files for frdm_k64f and
hexiwear_k64. They were incorrect in uart instance 5.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

galak committed:
pinmux: Move stm32 board pinmux files into board dirs

Move all STM32 based board pinmux files into the board dirs so we are
consistent across all the STM32 platforms/boards.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

liliu-l committed:
drivers: Add AES driver for rda_5981a, support ecb, cbc mode.

Signed-off-by: Li Liu <li.liu@linaro.org>

galak commented:
Closing out as this seems abandoned.
